ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -748,6 +748,11 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent keyEvent) {
                   new ReactTextInputSubmitEditingEvent(
                       editText.getId(),
                       editText.getText().toString()));
+              // if blurOnSubmit is false, and isMultiline, go to new line
```

Comment:
Hi shergin, thanks for the review.
as far as I know, the code should work just like you said:
```
blurOnSubmit && isMultiline => generate submit event; blur; return true;
!blurOnSubmit && isMultiline => return false; // Perform default behaviour.
blurOnSubmit && !isMultiline => generate submit event; blur; return true;
!blurOnSubmit && !isMultiline => return true; // Shallow <Enter>.
```
but I must admit that it is not very clear, I will try to rewrite the whole function with local variables